src/unix/linux_like/linux/gnu/b32/sparc/mod.rs

```diff
@@ -0,0 +1,976 @@
+//! SPARC-specific definitions for 32-bit linux-like values
+
+pub type c_char = u8;
+pub type wchar_t = i32;
+
+s! {
+    pub struct sigaction {
+        pub sa_sigaction: ::sighandler_t,
+        pub sa_mask: ::sigset_t,
+        pub sa_flags: ::c_int,
+        pub sa_restorer: ::Option<extern fn()>,
+    }
+
+    pub struct statfs {
+        pub f_type: ::__fsword_t,
+        pub f_bsize: ::__fsword_t,
+        pub f_blocks: ::fsblkcnt_t,
+        pub f_bfree: ::fsblkcnt_t,
+        pub f_bavail: ::fsblkcnt_t,
+
+        pub f_files: ::fsfilcnt_t,
+        pub f_ffree: ::fsfilcnt_t,
+        pub f_fsid: ::fsid_t,
+
+        pub f_namelen: ::__fsword_t,
+        pub f_frsize: ::__fsword_t,
+        f_spare: [::__fsword_t; 5],
+    }
+
+    pub struct siginfo_t {
+        pub si_signo: ::c_int,
+        pub si_errno: ::c_int,
+        pub si_code: ::c_int,
+        #[doc(hidden)]
+        #[deprecated(
+            since="0.2.54",
+            note="Please leave a comment on \
+                  https://github.com/rust-lang/libc/pull/1316 if you're using \
+                  this field"
+        )]
+        pub _pad: [::c_int; 29],
```

Comment:
This field should be private, and the `doc(hidden)` and deprecation not necessary since no code is using this.

Reply:
This struct is verbatim identical to several other architectures in this source, so I would rather commit this as-is and then make the requested changes later for all architectures at the same time.

I already spotted multiple smaller bugs in the sparc64 code which I wanted to fix after this has been merged and I'm happy to add your requested change to the list of fixes.

Would that be okay?

Reply:
> This struct is verbatim identical to several other architectures in this source, so I would rather commit this as-is and then make the requested changes later for all architectures at the same time.

Doing this change for other architectures would be a breaking change.

Reply:
Then I don't understand why it should be changed here now. It would be confusing for anyone reading the code.

Reply:
A bug was found that's hard to fix because it's an API breaking change.

If you want to argue that new APIs should also introduce the bug, please explain what is the value and cost of doing that, and while introducing the bug for newer APIs is a trade-off that is worth it.

Reply:
Well, if you had mentioned that there was a bug in the first place, I would have actually understood the argument.

I'll push something in a minute.

Reply:
Sorry, I'm very low on time. 

After this the rest looked good to me. I'll try to merge it ASAP but we are having network reliability problems.

Reply:
No worries. Pushed. Also fixed the names of two private "reserved" fields to match the actual glibc headers.